addons/web/static/tests/fields/one2many_field_tests.js

```diff
@@ -4527,17 +4530,29 @@ QUnit.module("Fields", (hooks) => {
                         </field>
                     </form>`,
                 resId: 1,
-                viewOptions: {
-                    mode: "edit",
-                },
             });
+            assert.verifySteps([
+                "X2ManyField readonly",
+                "CharField readonly",
+                "IntegerField readonly",
+            ]);
 
-            assert.strictEqual(editableWidgets, 1, "o2m is only widget in edit mode");
-            await click(form.$("tbody td.o_field_x2many_list_row_add a"));
-
-            assert.strictEqual(editableWidgets, 3, "3 widgets currently in edit mode");
+            await clickEdit(target);
+            assert.verifySteps([
+                "X2ManyField editable",
+                "CharField readonly",
+                "IntegerField readonly",
+            ]);
 
-            testUtils.mock.unpatch(AbstractField);
+            await click(target, "tbody td.o_field_x2many_list_row_add a");
+            assert.verifySteps([
+                "X2ManyField editable",
+                "X2ManyField editable",
+                "CharField readonly",
+                "IntegerField readonly",
+                "CharField editable",
+                "IntegerField editable",
+            ]);
```

Comment:
what about simply checking in the DOM? char and integer fields are inputs in edit mode. For the x2many, the test is not really relevant (what matters is the mode of the fields inside the x2many)